### PR TITLE
tests: add codecov config

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 
changed code may be not always able to covered by unit test

### What is changed and how it works?
add codecov config, mainly copy config from TiDB repo
pay attention to two fields
```
+  status:
+    project:
+      default:
+        threshold: 3 #Allow the coverage to drop by threshold%, and posting a success status.
+    patch:
+      default:
+        target: 0%
+    changes: no
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by amyangfei** [comment]
PTAL @GregoryIan @csuzhangxc

**Comment by GregoryIan** [comment]
LGTM

File: .codecov.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      macro: no
+
+comment:
+  layout: "header, diff"
+  behavior: default
+  require_changes: no
+
+ignore:
+  - "LICENSES"
+  - "*_test.go"
```
Comment (csuzhangxc) [inline .codecov.yml]: whether needing to ignore ".pb.go" files?

Comment (amyangfei) [reply .codecov.yml]: yep, can also add.
but we have already removed these files in build scripts

Comment (amyangfei) [reply .codecov.yml]: added

**Pull request merged**